contracts/GenArt721CoreV3.sol

```diff
@@ -223,11 +223,26 @@ contract GenArt721CoreV3 is
     /// default base URI to initialize all new project projectBaseURI values to
     string public defaultBaseURI;
 
+    modifier onlyNonZeroAddress(address _address) {
+        require(_address != address(0), "Must input non-zero address");
+        _;
+    }
+
+    modifier onlyNonEmptyString(string memory _string) {
+        require(bytes(_string).length != 0, "Must input non-empty string");
+        _;
+    }
+
     modifier onlyValidTokenId(uint256 _tokenId) {
         require(_exists(_tokenId), "Token ID does not exist");
         _;
     }
 
+    modifier onlyValidProjectId(uint256 _projectId) {
+        require(_projectId < _nextProjectId, "Project ID does not exist");
```

Comment:
nit: Since we start at a non-zero initial project ID, we could still accidentally set something for e.g. project 0. An alternate way to check if a project is valid would be to check if the project's artist address is non-zero, which I think is more robust (since we now never allow assigning a zero artist address in this PR).

Reply:
Good call out–that said, given that we also have reference to the `startingProjectId`, my inclination is to just bound check both of these, as this is a bit more idiomatic to read from my perspective. I'll update to do so, but please re-open if you disagree.

Reply:
ah, yes, I forgot about `startingProjectId`. SGTM!